internal/statesync/reactor.go

```diff
@@ -288,6 +291,7 @@ func (r *Reactor) Sync(ctx context.Context) (sm.State, error) {
 		r.stateProvider,
 		r.snapshotCh.Out,
 		r.chunkCh.Out,
+		r.snapshotCh.Done(),
```

Comment:
It looks like this is closed when the reactor's closeCh is called.
Any reason to not just rely on this behavior directly? 
https://github.com/tychoish/tendermint/blob/3dd0e3e125bfdf0b9acf7ceed04d070e2094439d/internal/statesync/reactor.go#L872

Reply:
Or, can we just pass through the entire `snapshotCh`?

Reply:
I tried a few rounds here, but the snapshot channel's lifecycle is managed by the router, not the reactor so using the reactors close channel results in a (possible, though unlikely) send into a closed channel.)